src/Square.java

```diff
@@ -15,6 +15,13 @@ public Square(int position, Piece occupyingPiece) {
 
     }
 
+    public Square(Square otherSquare) {
+        this.position = otherSquare.position;
+        this.occupyingPiece = new Piece(otherSquare.occupyingPiece);
```

Comment:
I think it's better/cleaner to use `getOccupyingPiece()` instead of accessing the field directly.